package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -316,9 +316,19 @@ def __init__(self, universe,
 
         self.u = universe
         self._trajectory = self.u.trajectory
-        self.donors_sel = donors_sel
-        self.hydrogens_sel = hydrogens_sel
-        self.acceptors_sel = acceptors_sel
+
+        self.donors_sel = donors_sel.strip() if donors_sel is not None else donors_sel
+        self.hydrogens_sel = hydrogens_sel.strip() if hydrogens_sel is not None else hydrogens_sel
+        self.acceptors_sel = acceptors_sel.strip() if acceptors_sel is not None else acceptors_sel
+
+        msg = ("{} is an empty selection string - no hydrogen bonds will "
```

Comment:
I'll be honest - I'm still not sure in what scenario this would be a case that would be expected.

However - in the interest of getting this done I'll merge this but keep the underlying issue open. We can discuss this further there.

Reply:
Yeah that's true. If there's an empty selection string then there's no point in running the analysis - no hydrogen bonds can possible be found...